pkg/apis/k0s/v1beta1/elb.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2024 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+// Defaults are keepalived's defaults.
+const defaultVirtualRouterID = 51
+const defaultAdvertInterval = 1
+const defaultInterface = "eth0"
+
+// ExternalLoadBalancingSpec defines the configuration options related to k0s's
+// keepalived feature.
+type ExternalLoadBalancingSpec struct {
```

Comment:
I'm not very happy with the ExternalLB/ExternalLoadBalancing names. It hints that it's running outside. Also it's contraction to ELB can be confused for AWS ELB which is something that I, again, dislike.

In the first implementation I called it keepalived, which works really well, however I really like that NLLB API allows of not calling it keepalived in case the software used for this changes.

I've considered  ControlPlaneLoadBalancing or CPLB, I'd like to hear opinions on this.

Reply:
Renamed it to ControlPlaneLoadBalancing and I like it much more but I'm still not 100% convinced